umh-core/pkg/service/connection/connection.go

```diff
@@ -0,0 +1,510 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package connection provides a service for managing and monitoring network
+// connectivity to remote assets using Nmap as the underlying probe mechanism.
+//
+// # Key Concepts
+//
+// - Connections have multiple health indicators (running, reachable, flaky)
+// - Flakiness detection requires multiple samples over time
+// - The service translates network probe results into business-relevant health status
+package connection
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config/connectionconfig"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config/nmapserviceconfig"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/constants"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/logger"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/filesystem"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/nmap"
+	"go.uber.org/zap"
+)
+
+// ServiceInfo holds information about the connection's health status.
+// It uses separate boolean flags for different health aspects instead of
+// a single status value, making it easier to check specific conditions.
+type ServiceInfo struct {
+	// IsRunning indicates if the underlying Nmap service is running.
+	// This refers to the monitoring service itself, not the target endpoint.
+	IsRunning bool
```

Comment:
IsRunning, PortStateOpen, IsReachable, IsFlaky would make an good enum.
 
enum (
 Down = iota
 Running
 Reachable
 Flaky
)

(PortStateOpen is not needed here, as there shouldnt be any situation where IsRunning == false && PortStateOpen == true).

Reply:
also thought about that, and even added it to the comments. but then decided to follow the AI approach and not use an enum and instead multiple boolean flags.

the resulting checks could then be made much easier with IsPortStateOpen && !IsNotFlaky

this could be a little bit better readable as if I were to do it here in an enum